src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -65,13 +65,17 @@ def tgt_closure_platforms(tgts):
   :returns: a dict mapping a platform string to a list of targets that specify the platform.
   """
   tgts_by_platforms = {}
-  for tgt in tgts:
+
+  def insert_or_append_tgt_by_platform(tgt):
     if tgt.platforms:
       for platform in tgt.platforms:
         if platform in tgts_by_platforms:
           tgts_by_platforms[platform].append(tgt)
         else:
           tgts_by_platforms[platform] = [tgt]
+
+  for tgt in tgts:
+    insert_or_append_tgt_by_platform(tgt)
```

Comment:
if you mean to use adopt the functional style like `filter(predicate, tgts)`, you can probably use
`map(insert_or_append_tgt_by_platform, tgts)`?